course7/homeworks/anton/ls/main.cc

```diff
@@ -0,0 +1,135 @@
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+
+namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+
+struct configuration
+{
+    std::string file_path = ".";
+    bool show_help = false;
+    bool show_tree = false;
+    bool show_long = false;
+};
+
+configuration parse_args(int argc, char *argv[]);
+void print_file_tree(const fs::path& dir, const configuration& config, const std::string& filler = "");
+size_t get_file_size(const fs::path&);
+
+int main(int argc, char *argv[])
+{
+    try
+    {
+        const configuration& config = parse_args(argc, argv);
+
+        if (!config.show_help)
+        {
+            fs::path path(config.file_path);
+            print_file_tree(path, config);
+        }
+
+        return 0;
+    }
+    catch(std::exception& ex)
+    {
+        std::cout << "\033[1;31m" << ex.what() << "\033[0m" << std::endl;
```

Comment:
Don't use color unless you make sure first that the output is a console. If the output is redirected to a file, or piped to another program for parsing, this will only create a mess.
When in doubt, don't use them at all.

Reply:
fixed